### PR TITLE
tests: Added missing tests for RPC wallet errors

### PR DESCRIPTION
**Pull request by benthecarman**

### CODE REVIEW

File: test/functional/wallet_basic.py [hunk 1]
```diff
@@ -327,12 +327,38 @@ def run_test(self):
...
+        # This will raise an execption for attempting to set a label for an invalid Bitcoin address
+        assert_raises_rpc_error(-5, "Invalid Bitcoin address", self.nodes[0].setlabel, "invalid address", "label")
+
+        # This will raise an execption for importing an invalid address
+        assert_raises_rpc_error(-5, "Invalid Bitcoin address or script", self.nodes[0].importaddress, "invalid")
+
+        # This will raise an execption for attempting to import a pubkey that isn't in hex
+        assert_raises_rpc_error(-5, "Pubkey must be a hex string", self.nodes[0].importpubkey, "not hex")
+
+         # This will raise an execption for importing an invalid pubkey
```
Comment (fanquake) [inline test/functional/wallet_basic.py]: Please fix the indentation here. https://travis-ci.org/bitcoin/bitcoin/jobs/491122519

**Comment by practicalswift** [comment]
Concept ACK

File: test/functional/wallet_basic.py [hunk 2]
```diff
@@ -327,12 +327,38 @@ def run_test(self):
...
+        assert_raises_rpc_error(-5, "Invalid private key encoding", self.nodes[0].importprivkey, "invalid")
+
+        # This will raise an exception for importing an address with the PS2H flag
+        temp_address = self.nodes[1].getnewaddress()
+        assert_raises_rpc_error(-5, "Cannot use the p2sh flag with an address - use a script instead", self.nodes[0].importaddress, temp_address, "label", False, True)
+
+        # This will raise an exception for attempting to dump the private key of an address you do not own
+        assert_raises_rpc_error(-3, "Address does not refer to a key", self.nodes[0].dumpprivkey, temp_address)
+
+        # This will raise an execption for attempting to get the private key of an invalid Bitcoin address
```
Comment (practicalswift) [inline test/functional/wallet_basic.py]: Execption is an exceptional spelling of exception :-)

Applies also to the four execptional cases below.

Comment (benthecarman) [reply test/functional/wallet_basic.py]: Fixed

**Comment by promag** [comment]
utACK dc3b2cc.

File: test/functional/wallet_basic.py [hunk 3]
```diff
@@ -327,12 +327,38 @@ def run_test(self):
...
 
         # This will raise an exception since generate does not accept a string
         assert_raises_rpc_error(-1, "not an integer", self.nodes[0].generate, "2")
 
+        # This will raise an exception for the invalid private key format
+        assert_raises_rpc_error(-5, "Invalid private key encoding", self.nodes[0].importprivkey, "invalid")
+
+        # This will raise an exception for importing an address with the PS2H flag
+        temp_address = self.nodes[1].getnewaddress()
+        assert_raises_rpc_error(-5, "Cannot use the p2sh flag with an address - use a script instead", self.nodes[0].importaddress, temp_address, "label", False, True)
```
Comment (MarcoFalke) [inline test/functional/wallet_basic.py]: nit in the future use named arguments

**Comment by MarcoFalke** [comment]
Also reminds me that we shouldn't extend wallet_basic, but rather extend the more specific wallet tests

**Pull request merged**